### PR TITLE
Added feature to check if a reference is contained within another reference

### LINKED ISSUE #5: Add functionality to check if a specific Bible reference belongs to a general range of references
Looking for a function that takes two references (a specific chapter:verse reference and a more general one) and returns true if they are related.

i.e.
```
belongsToRef({ ref: "1:3-15; 2:5-10,24", check: "2:8"}) // returns true
belongsToRef({ ref: "1:3-15; 2:5-10,24", check: "1:16"}) // returns false
```

ALSO:
There are some [common verse number](https://pursuitbible.com/bible-verse-number-references/) references that are not implemented that we would like to implement, primarily:

- An entire chapter (i.e 6)
- A range of chapters (i.e 1-3)
- All the verses following (i.e 2:11 ff)

Could we add the range: 1-3, meaning chapter 1 to 3. On the filter, it would match any chapter greater than 1 and less than 3, and any verse greater than 1. So 1:4 would be included in that range so it would return true also would 3:9999

### PR DESCRIPTION
**Pull request by kintsoogi**
Added feature to check if a reference is contained within another reference

- Code for this feature is contained in `helpers/doesReferenceContain.js`

- Tests of this function are contained in `referenceHelpers.test.js` under the describe function `Test doesReferenceContain`.

- Documentation of the new function (and of all exposed fucntions) has been added via jsDoc & docdash. This can be viewed at the [draft netlify site](https://vermillion-custard-b8702a.netlify.app/)

### CODE REVIEW

**Comment by kintsoogi** [comment]
@abelpz @PhotoNomad0 Should be ready to be tested and reviewed! No rush, just wanted to get this out here before leaving for All Staff`

**Comment by abelpz** [comment]
Thanks @kintsoogi! I won't be at the all staff, I can review it

**Comment by kintsoogi** [comment]
Sweet, oh and @abelpz @PhotoNomad0 [here is a little codesandbox](https://codesandbox.io/s/bible-reference-range-72yvdu?file=/src/App.js )if that helps to test it quickly!

**Comment by abelpz** [comment]
@kintsoogi I found one case that doesn't work as I would expect:

reference: 3:1-10
lookup: 3:1,11
strict: true

I would expect this to return false since 3:11 is not included in 3:1-10

**Comment by PhotoNomad0** [comment]
> @kintsoogi I found one case that doesn't work as I would expect:
> 
> reference: 3:1-10 lookup: 3:1,11 strict: true
> 
> I would expect this to return false since 3:11 is not included in 3:1-10

It looks like you two have greatly expanded the capabilities of this API.

**Comment by kintsoogi** [comment]
@abelpz @PhotoNomad0 Alright I fixed the logic for the problem case that Abel talked about! Thanks for catching that by the way! Ready to review whenever y'all get the chance.

[Documentation](https://vermillion-custard-b8702a.netlify.app/)

[Code Sandbox](https://codesandbox.io/s/bible-reference-range-72yvdu?file=/src/App.js)

**Comment by abelpz** [comment]
Thank you @kintsoogi, this is working great! 🚀

File: src/helpers/referenceHelpers.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const RANGE_SEPARATORS = [
  * takes a reference and splits into individual verses or verse spans.
  * @param {string} ref - reference in format such as:
  *   “2:4-5”, “2:3a”, “2-3b-4a”, “2:7,12”, “7:11-8:2”, "6:15-16;7:2"
- * @return {[{chapter, verse, endChapter, endVerse}]}
+ * @return {Object}  The Verse Chunk returned
```
Comment (PhotoNomad0) [inline src/helpers/referenceHelpers.js]: Please investigate other ways to show more detail than just object type

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]

File: src/helpers/referenceHelpers.js (hunk 1 above)
Comment (abelpz) [reply src/helpers/referenceHelpers.js]: > Please investigate other ways to show more detail than just object type

I guess this has to do with some limitation with the jsdoc API docs generator, which I have encountered before. If that's the case, what I do is create a typedef and use that to define the type of the return: https://github.com/Proskomma/epitelete/blob/f2d20d5cd30270920b053c00c796df68f93b20b5/src/index.js#L622

That will make the generator not get confused.

Comment (kintsoogi) [reply src/helpers/referenceHelpers.js]: Thanks Abel that was a great help! I went ahead and made that change

**Review by PhotoNomad0 — APPROVED** [review]
LGTM

**Pull request merged**